agent/task.go

```diff
@@ -149,7 +149,7 @@ func (a *Agent) runPreTaskCommands(ctx context.Context, tc *taskContext) {
 			return
 		}
 		if taskGroup.SetupGroup != nil {
-			err = a.runCommands(ctx, tc, taskGroup.SetupGroup.List(), false)
+			err = a.runCommands(ctx, tc, taskGroup.SetupGroup.List(), taskGroup.SetupGroupFailTask)
```

Comment:
By using the existing parameter isTaskCommands for this, we try setCurrentCommand and setCurrentTimeout when isTaskCommands (i.e. SetupGroupFailTask) is true. Will this cause any issues if we still do this anyway? Otherwise we may want to change this parameter to accept something besides a boolean so we can have multiple options, or add a parameter.

Reply:
Are you talking about isTaskCommands? I don't think we can remove it entirely because it sets important state in the agent

Reply:
I assumed he meant only running setCurrentCommand and setCurrentTimeout when isTaskCommands is true, might be wrong.

Reply:
Sorry, by "that branch" I meant https://github.com/evergreen-ci/evergreen/blob/master/agent/command.go#L92-L98.

Reply:
What's the change you want to make? Remove the else branch?

Reply:
Yeah, what if we always did the part in the `if`, and never the part in the `else`.

Reply:
I think that makes sense, @john-m-liu any objections?

Reply:
Looks like we can't remove the branch, and additionally we can't rely on isTaskCommands when we want setup_group to fail because it will update these parameters when we do not want to, so I'll be adding a new parameter.